lib/editor/components/map/LocationsLayer.js

```diff
@@ -0,0 +1,229 @@
+          layer.coordSet.length &&
+          typeof layer.coordSet.length === 'number' &&
+          layer.coordSet.length > 0
+      )
+      .map((layer) => {
+        // $FlowFixMe flow doesn't understand filter
+        const { _leaflet_id: leafletId, _latlngs: coordSet } = layer
+
+        const isPolygon = layer instanceof L.Polygon
+        // FIXME: support all layer types once all layer types can be created in editor
+        // via switch statement or similar
+        layerType = isPolygon ? 'polygon' : 'polyline'
+
+        // Polygons have 2D latlng object structure
+        const layerCoords = isPolygon
+          ? coordSet[0].map(convertLatLngToArray)
+          : coordSet.map(convertLatLngToArray)
+
+        // If dealing with a shape fromSaved, the editcontrol has changed the leaflet ID
+        // TODO: fix this more elegantly w/ the forked react-leaflet-draw repo here: https://www.npmjs.com/package/@andrewdodd/react-leaflet-draw
+        // (which requires React 16) or use refs.
+        // Create our locations shape out of the bulk location shape points
+        // TODO: create a function for this since it's shared with onEdited?
+        const groupedLocationShapePts: GroupedLocationShapePts = groupLocationShapePoints(
+          locationShapes
+        )
+
+        // Get the geometry_ids that match the deleted shape
+        const onDeletedKeys: Array<string> = Object.keys(
+          groupedLocationShapePts
+        ).filter((key) => {
+          if (key === leafletId) return false
+          if (this.checkSameShape(groupedLocationShapePts[key], layerCoords)) { return false }
+          return true
+        })
+
+        // Filter deleted geometry_ids out of the bulk locationShapes data
+        locationShapes = locationShapes.filter((shapePt) => {
+          if (onDeletedKeys.includes(shapePt.geometry_id)) return false
+          return true
+        })
+      })
+
+    if (layerType) this.props.updatingLocationShape({ geometryType: layerType, shapes: locationShapes })
+  }
+
+  _onEdited (e: L.LeafletEvent) {
+    const {activeEntity} = this.props
+    const {layers: { _layers: layers }} = e
+
+    let layerType = null
+    let locationShapes = clone(activeEntity.location_shapes)
+    Object.values(layers)
+      .filter(layer => !!layer && typeof layer === 'object' && layer.hasOwnProperty('editing') && layer.hasOwnProperty('_leaflet_id') && layer.hasOwnProperty('_latlngs'))
+      .map((layer: L.Polygon | L.Polyline) => {
+      // Do a manual and complicated type check
+        const {editing} = layer
+
+        const isPolygon = layer instanceof L.Polygon
+        // FIXME: support all layer types once all layer types can be created in editor
+        // via switch statement or similar
+        layerType = isPolygon ? 'polygon' : 'polyline'
+
+        const formattedEditedCoords = isPolygon
+          ? editing.latlngs[0][0].map(convertLatLngToArray)
+          : editing.latlngs[0].map(convertLatLngToArray)
```

Comment:
Referencing this earlier comment, https://github.com/ibi-group/datatools-ui/pull/754#discussion_r799807444, extract this function, and reuse (see around line 130, and I think it may appear in other files too):
```js
const formattedEditedCoords = getCoordinates(layer, editing.latlngs[0]).map(...)
...
function getCoordinates(layer: L.Polygon | L.Polyline, coordinates: Array<...>): Array<...> {
  const isPolygon = ...
  return isPolygon ? coordinates[0] : coordinates
}
```